### PR TITLE
Add Camel AS2 Modules as Quarkus Extension #1471

### PR DESCRIPTION
**Pull request by JiriOndrusek**
Issue: https://github.com/apache/camel-quarkus/issues/1471

As a part of the solution `extension-support/bouncycastle` was created. Bouncy castle security provider is required for camel-as2 and without proper initialization there are no digests working with quarkus in native.

[ ] An issue should be filed for the change unless this is a trivial change (fixing a typo or similar). One issue should ideally be fixed by not more than one commit and the other way round, each commit should fix just one issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful and properly spelled subject line and body. Copying the title of the associated issue is typically enough. Please include the issue number in the commit message prefixed by #.
[ ] The pull request description should explain what the pull request does, how, and why. If the info is available in the associated issue or some other external document, a link is enough.
[ ] Phrases like Fix #<issueNumber> or Fixes #<issueNumber> will auto-close the named issue upon merging the pull request. Using them is typically a good idea.
[ ] Please run mvn process-resources -Pformat (and amend the changes if necessary) before sending the pull request.
[ ] Contributor guide is your good friend: https://camel.apache.org/camel-quarkus/latest/contributor-guide.html

### CODE REVIEW

**Comment by oscerd** [comment]
@JiriOndrusek since you have some experience with as2, would you mind creating also an example of usage in camel-examples? Thank you

**Comment by JiriOndrusek** [comment]
@oscerd I'll look into it.

File: extensions-support/bouncycastle/runtime/pom.xml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        <artifactId>camel-quarkus-support-bouncycastle-parent</artifactId>
+        <version>1.1.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>camel-quarkus-support-bouncycastle</artifactId>
+    <name>Camel Quarkus :: Support :: Bouncy Castle :: Runtime</name>
+
+    <properties>
+        <firstVersion>1.0.0-M6</firstVersion>
```
Comment (jamesnetherton) [inline extensions-support/bouncycastle/runtime/pom.xml:33]: Should be `1.0.0-CR4`.

Comment (JiriOndrusek) [reply extensions-support/bouncycastle/runtime/pom.xml:33]: I've created this file manually and missed that one, I' fix it

**Comment by JiriOndrusek** [comment]
Change in camel - https://github.com/apache/camel/pull/4031 - fixes failed tests

**Comment by JiriOndrusek** [comment]
There seems to be more issues to solve. I'm closing for now.

**Pull request closed without merging**